servers/src/main/java/tachyon/master/next/filesystem/FileSystemMaster.java

```diff
@@ -461,6 +463,68 @@ public FileBlockInfo getFileBlockInfo(long fileId, int fileBlockIndex)
   }
 
   /**
+   * Return whether the file is fully in memory or not. The file is fully in memory only if all the
+   * blocks of the file are in memory, in other words, the in memory percentage is 100.
+   *
+   * @return true if the file is fully in memory, false otherwise
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public boolean isFullyInMemory(TachyonURI path)
+      throws FileDoesNotExistException, InvalidPathException {
+    return getInMemoryPercentage(path) == 100;
+  }
+
+  /**
+   * Get the in-memory percentage of an InodeFile determined by path. For a file that has all blocks
+   * in memory, it returns 100; for a file that has no block in memory, it returns 0.
+   *
+   * @param path path to the file
+   * @return the in memory percentage
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public int getInMemoryPercentage(TachyonURI path)
+      throws FileDoesNotExistException, InvalidPathException {
+    long fileId = getFileId(path);
+    Inode inode = mInodeTree.getInodeById(fileId);
+    if (inode == null) {
+      throw new FileDoesNotExistException(path + " does not exist.");
+    }
+    if (!inode.isFile()) {
+      throw new FileDoesNotExistException(path + " is not a file.");
+    }
+    InodeFile inodeFile = (InodeFile) inode;
+
+    long length = inodeFile.getLength();
+    if (length == 0) {
+      return 100;
+    }
+
+    long inMemoryLength = 0;
+    for (BlockInfo info : mBlockMaster.getBlockInfoList(inodeFile.getBlockIds())) {
+      if (isInMemory(info)) {
+        inMemoryLength += info.getLength();
+      }
+    }
+    return (int) (inMemoryLength * 100 / inMemoryLength);
+  }
+
+  /**
+   * @return true if the given block is in some worker's memory, false otherwise
+   */
+  private boolean isInMemory(BlockInfo blockIfno) {
+    for (BlockLocation location : blockIfno.getLocations()) {
+      int tier = location.getTier();
+      int storageLevelValue = StorageDirId.getStorageLevelAliasValue(tier);
```

Comment:
I think getStorageLevelAliasValue() takes in a storage dir id, and extracts the tier from it. However, I *think* location.getTier() is already the tier, and not the storage dir id.